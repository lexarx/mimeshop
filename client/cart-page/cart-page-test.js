jest.mock('api-request');
import React from 'react'
import CartPage from './cart-page'
import { mount } from 'enzyme'
import apiRequest from 'api-request'

test('displays cart page', async () => {
	apiRequest.mockImplementation((method, url) => {
		if (method === 'POST' && url === 'cart') {
			return {
				items: [
					{
						product: {
							id: '1',
							shortDescription: `Short description 1`,
							preview: 'http://via.placeholder.com/100?text=P1',
							price: 100.00
						},
						quantity: 1
					},
					{
						product: {
							id: '2',
							shortDescription: `Short description 2`,
							preview: 'http://via.placeholder.com/100?text=P2',
							price: 100.00
						},
						quantity: 2
					}
				],
				total: 300.00
			};
		}
	});
	const cartPage = mount(<div><CartPage/></div>);
	await new Promise(resolve => setTimeout(resolve, 0));
	cartPage.update();
	expect(cartPage).toMatchSnapshot();
});

test('navigates to product page', async () => {
	apiRequest.mockImplementation((method, url) => {
		if (method === 'POST' && url === 'cart') {
			return {
				items: [
					{
						product: {
							id: '1',
							shortDescription: `Short description 1`,
							preview: 'http://via.placeholder.com/100?text=P1',
							price: 100.00
						},
						quantity: 1
					}
				],
				total: 100.0
			};
		}
	});
	const navigate = jest.fn();
	const cartPage = mount(<div><CartPage navigate={navigate}/></div>);
	await new Promise(resolve => setTimeout(resolve, 0));
	cartPage.update();
	cartPage.find('img').simulate('click');
	expect(navigate).toHaveBeenCalledWith('/products/1');
});