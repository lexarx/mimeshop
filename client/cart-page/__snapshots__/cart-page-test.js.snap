// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays cart page 1`] = `
<div>
  <CartPage>
    <div
      className="product"
      key="1"
    >
      <img
        className="image"
        onClick={[Function]}
        src="http://via.placeholder.com/100?text=P1"
      />
      <div
        className="shortDescription"
        onClick={[Function]}
      >
        Short description 1
      </div>
      <div
        className="quantity"
      >
        Quantity: 
        1
      </div>
    </div>
    <div
      className="product"
      key="2"
    >
      <img
        className="image"
        onClick={[Function]}
        src="http://via.placeholder.com/100?text=P2"
      />
      <div
        className="shortDescription"
        onClick={[Function]}
      >
        Short description 2
      </div>
      <div
        className="quantity"
      >
        Quantity: 
        2
      </div>
    </div>
    <div
      className="total"
    >
      Total price: 
      300
    </div>
    <button
      className="checkoutButton"
      onClick={[Function]}
    >
      Checkout
    </button>
  </CartPage>
</div>
`;
